File: Backend/src/modules/tooltip/domain_value/item_tooltip.rs
use crate::modules::tooltip::domain_value::{Stat, ItemSet, Socket};

#[derive(Debug, Clone, Serialize, JsonSchema)]
pub struct ItemTooltip {
  pub name: String,
  pub icon: String,
  pub quality: u8,
  pub bonding: String,
  pub inventory_type: String,
  pub sheath_type: String,
  pub sub_class: String,
  pub armor: Option<u16>,
  pub stats: Vec<Stat>,
  pub durability: Option<u16>,
  pub item_level: Option<u16>,
  pub required_level: Option<u16>,
  pub item_effects: Vec<String>,
  pub item_set: Option<ItemSet>,
  pub socket: Option<Socket>,
  pub enchant: Option<String>
}

File: Webclient/src/app/module/tooltip/module/item_tooltip/domain_value/item_tooltip.ts
export interface ItemTooltip {
    item_id: number;
    name: string;
    icon: string;
    quality: number;
    inventory_type: string | undefined;
    bonding: string | undefined;
    sheath_type: string | undefined;
    sub_class: string;
    armor: number | undefined;
    stats: Array<{ value: number, name: string }> | undefined;
    durability: number | undefined;
    item_level: number | undefined;
    required_level: number | undefined;
    item_effects: Array<string> | undefined;
    item_set: {
        name: string,
        set_items: Array<{ item_id: number, active: boolean, name: string, item_level: number, inventory_type: number }>,
        set_effects: Array<{ threshold: number, description: string }>
    } | undefined;
    socket: { socket_bonus: string, slots: Array<{ flag: number, item: { icon: string, effect: string, flag: number } | undefined }> } | undefined;
    enchant: string | undefined;
    weapon_stat: { delay: number, damage_sources: Array<{ damage_min: number, damage_max: number, damage_type: string | undefined }> } | undefined;
}
